pkg/status/status.go

```diff
@@ -65,9 +65,30 @@ func (s *Service) calculateTestStatus(tr v1alpha1.TestResult, maxRetries, count
 		return v1alpha1.TestNotYetScheduled
 	}
 
-	// TODO handle maxRetries https://github.com/kyma-incubator/octopus/issues/8
 	if maxRetries > 0 {
-		return v1alpha1.TestUnknown
+		var anySucceeded bool
+		var anyRunning bool
+		for _, exec := range tr.Executions {
```

Comment:
_just an idea:_ 
you have similar logic in `pkg/scheduler/retry.go` **maybe** (I'm not convinced) it will be worth to extract to some method and just reuse it in `pkg/scheduler/retry.go`.